File: src/LinksNews/angular2App/app/core/domain/page/link.ts
import { ViewModes } from "../../common/enums";
import * as _ from 'lodash';

export class Link {
    id: number;
    columnId: number;
    linkIndex: number;
    title: string = "New Link";
    description: string = "New Link Description";
    imageUrl: string;
    hint: string;
    href: string = "http://#";
    buttonAccess: boolean;
    buttonTitle: string;
    buttonIndex: number;
    buttonImageUrl: string;
    showImage: boolean = true;
    showDescription: boolean = true;
    showAd: boolean = false;
    adContent: string;
    newsLink: boolean = false;
    viewModeId: ViewModes = ViewModes.List;
    selected: boolean = false;

    clone(): Link {
        let result: Link = <Link>_.assign(new Link(), this);
        return result;
    }

    static from(link: Link): Link {
        let result: Link = new Link();
        _.assign(result, link);
        return result;
    }
}

File: src/LinksNews/Services/Data/Models/Link.cs
using LinksNews.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace LinksNews.Services.Data.Models
{
    [Table("link")]
    public class Link
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long? Id { get; set; }
        public long? ColumnId { get; set; }
        public long? LinkIndex { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Hint { get; set; }
        public string Href { get; set; }
        public bool ButtonAccess { get; set; }
        public string ButtonTitle { get; set; }
        public long? ButtonIndex { get; set; }
        public bool ShowImage { get; set; }
        public bool ShowDescription { get; set; }
        public bool NewsLink { get; set; }
        public ViewModes? ViewModeId { get; set; }

        [NotMapped]
        public string ButtonImageUrl { get; set; }

        [NotMapped]
        public string ImageUrl { get; set; }

        [NotMapped]
        public bool ShowAd { get; set; }

        [NotMapped]
        public string AdContent { get; set; }
    }
}
